zerver/tests/test_messages.py

```diff
@@ -2386,6 +2387,22 @@ def test_jabber_mirror(self) -> None:
         self.assertTrue(bob.is_mirror_dummy)
 
 class MessageAccessTests(ZulipTestCase):
+    def test_update_invalid_flags(self) -> None:
+        message = self.send_personal_message(
+            self.example_email("cordelia"),
+            self.example_email("hamlet"),
+            "hello",
+        )
+        user_profile = self.example_user('hamlet')
+        for first_non_api_flag in UserMessage.NON_API_FLAGS:
+            break
+
+        with self.assertRaises(JsonableError):
+            do_update_message_flags(user_profile, get_client("website"), 'remove', first_non_api_flag, [message])
+
+        with self.assertRaises(JsonableError):
+            do_update_message_flags(user_profile, get_client("website"), 'remove', 'invalid', [message])
```

Comment:
In our tests, we usually prefer API interaction to calling raw `actions.py` methods where possible.  I'll just redo this that way.

Reply:
```
         )
-        user_profile = self.example_user('hamlet')
-        for first_non_api_flag in UserMessage.NON_API_FLAGS:
-            break
 
-        with self.assertRaises(JsonableError):
-            do_update_message_flags(user_profile, get_client("website"), 'remove', first_non_api_flag, [message])
+        self.login(self.example_email("hamlet"))
+        result = self.client_post("/json/messages/flags",
+                                  {"messages": ujson.dumps([message]),
+                                   "op": "add",
+                                   "flag": "invalid"})
+        self.assert_json_error(result, "Invalid flag: 'invalid'")
 
-        with self.assertRaises(JsonableError):
-            do_update_message_flags(user_profile, get_client("website"), 'remove', 'invalid', [message])
+        result = self.client_post("/json/messages/flags",
+                                  {"messages": ujson.dumps([message]),
+                                   "op": "add",
+                                   "flag": "is_private"})
+        self.assert_json_error(result, "Invalid flag: 'is_private'")
+
+        result = self.client_post("/json/messages/flags",
+                                  {"messages": ujson.dumps([message]),
+                                   "op": "add",
+                                   "flag": "active_mobile_push_notification"})
+        self.assert_json_error(result, "Invalid flag: 'active_mobile_push_notification'")
 
     def change_star(self, messages: List[int], add: bool=True, **kwargs: Any) -> HttpResponse:
```